### DISCUSSION TITLE
Memory leak with getters in Pinia

### DISCUSSION THREAD

**Discussion by vhema97** [vuejs/pinia#2689] [Help and Questions]
I'm using Options API in SSR app and have a root store defined :

app.js 
```
import { useRootStore } from './store';

const pinia = createPinia();
app.use(pinia);
const rootStore = useRootStore(pinia);
```

rootStore.js 

```
export const useRootStore = defineStore('root', {
   state: () => ({
    common: {
       settings: {},
    }
  }),
  getters: {
    getSpecificSetting (state) {
       return someMethod.getSetting(state.common.settings, 'app_setting');
    },
    isAppSettingsEnabled () {
       return this.getSpecificSetting.enabled;
    }
  }
}
```

Accessing the getter - getSpecificSetting in a component or in another getter is causing memory leak. state.common.settings data is of size 195KB.
**Version used :**
vue - 3.3.4
pinia - 2.1.7

**Comment by kevinramharak** [comment]
What is `someMethod` in this context? without a full reproduction it will be impossible to determine if there even is an issue here.
